rasa/core/channels/console.py

```diff
@@ -16,26 +17,16 @@
 from rasa.core.constants import DEFAULT_SERVER_URL
 from rasa.core.interpreter import INTENT_MESSAGE_PREFIX
 from rasa.utils.io import DEFAULT_ENCODING
-from typing import Any, Coroutine
 
 logger = logging.getLogger(__name__)
 
 DEFAULT_STREAM_READING_TIMEOUT_IN_SECONDS = 10
 
 
 def print_bot_output(
-    message: Dict[Text, Any], color=cli_utils.bcolors.OKBLUE
+    message: Dict[Text, Any], is_last=False, color=cli_utils.bcolors.OKBLUE
 ) -> Optional[questionary.Question]:
-    if ("text" in message) and not ("buttons" in message):
-        cli_utils.print_color(message.get("text"), color=color)
-
-    if "image" in message:
-        cli_utils.print_color("Image: " + message.get("image"), color=color)
-
-    if "attachment" in message:
-        cli_utils.print_color("Attachment: " + message.get("attachment"), color=color)
-
-    if "buttons" in message:
+    if "buttons" in message and is_last:
```

Comment:
Did you consider putting this "if" condition and the on line 50 together? Or maybe even extract button processing to a separate method, since it's pretty isolated functionality, that doesn't use any "state".
IMO that would improve readability.

Reply:
Extracted button printing into `print_buttons` function